lib/Draggable.es6

```diff
@@ -138,6 +138,13 @@ export default class Draggable extends DraggableCore {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
Moveonstartchange has been removed.

You can either use DraggableCore, or attach a ref to the Draggable el and
call setState on it directly.
On Jan 5, 2016 12:47 PM, "josdeweger" <notifications@github.com> wrote:

> In lib/Draggable.es6
> <https://github.com/mzabriskie/react-draggable/pull/127#discussion_r48880123>
> :
>
> > @@ -138,6 +138,13 @@ export default class Draggable extends DraggableCore {
> >      }
> >    }
> >
> > +  componentWillReceiveProps(nextProps) {
>
> Hm good point. Any suggestions on how to solve this issue?
>
> I have the following setup:
>
> Component <Lane> contains multiple <Batch>
>
> export default class Lane extends React.Component {
>   [...]
>
>   constructor(props) {
>     super(props);
>
>     this.state = {
>       visibleLaneWidth: 0
>     };
>   }
>
>   componentDidMount() {
>     this.setState({
>       visibleLaneWidth: this.refs.laneContent.offsetWidth
>     });
>   }
>
>   render() {
>     [...]
>     const { visibleLaneWidth } = this.state;
>
>     return (
>       [...]
>       <div ref="laneContent">
>           {batches.map((batch) => {
>             return (
>               <Batch key={batch.id}
>                     id={batch.id}
>                     title={batch.title}
>                     status={batch.status}
>                     period={batch.period}
>                     visiblePeriod={visiblePeriod}
>                     visibleLaneWidth={visibleLaneWidth} />
>             );
>           })}
>       </div>
>     );
>   }
> }
>
> Component <Batch> contains a <Draggable>:
>
> export default class Batch extends React.Component {
>    calculatePositionX() {
>      var visiblePeriodStartDate = this.props.visiblePeriod.start;
>      var visibleNrOfDays = this.props.visiblePeriod.diff('days');
>      var diffStartDateBatchStartDateVisiblePeriod = this.props.period.start.diff(visiblePeriodStartDate, 'days');
>      var xPos = (diffStartDateBatchStartDateVisiblePeriod / visibleNrOfDays) * this.props.visibleLaneWidth;
>
>      return xPos;
>   }
>
>   render() {
>       [...]
>       const posX = this.calculatePositionX();
>       const posY = 0;
>       [...]
>
>       return (
>          <span>
>            <Draggable
>              axis="x"
>              start={{x: posX, y: posY}}
>              moveOnStartChange={true} grid={grid}>
>              <div />
>            </Draggable>
>          </span>
>        );
>     }
> }
>
> The method calculatePositionX depends on the prop visibleLaneWidth, which
> is at first 0, so the initial outcome of the method is also 0. As soon as
> the visibleLaneWidth changes the method gets called again and returns the
> correct number, however, the Draggable is not updated and stays at x
> position 0. You can see this failing by executing the test in this PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mzabriskie/react-draggable/pull/127/files#r48880123>.
>